src/getDataDump.ts

```diff
@@ -75,81 +81,96 @@ export default async function getDataDump(
         }
     }
 
-    // to avoid having to load an entire DB's worth of data at once, we select from each table individually
-    // note that we use async/await within this loop to only process one table at a time (to reduce memory footprint)
-    while (tables.length > 0) {
-        const table = tables.shift()!
-
-        if (table.isView && !options.includeViewData) {
-            // don't dump data for views
-            retTables.push(merge<Table>([table, {
-                data: null,
-            }]))
+    try {
 
-            // eslint-disable-next-line no-continue
-            continue
+        if(options.lockTables) {
+            // see: https://dev.mysql.com/doc/refman/5.7/en/replication-solutions-backups-read-only.html
+            await executeSql(connection, `FLUSH TABLES WITH READ LOCK`)
+            await executeSql(connection, `SET GLOBAL read_only = ON`)
         }
 
-        currentTableLines = options.returnFromFunction ? [] : null
+        // to avoid having to load an entire DB's worth of data at once, we select from each table individually
+        // note that we use async/await within this loop to only process one table at a time (to reduce memory footprint)
+        while (tables.length > 0) {
+            const table = tables.shift()!
+
+            if (table.isView && !options.includeViewData) {
+                // don't dump data for views
+                retTables.push(merge<Table>([table, {
+                    data: null,
+                }]))
+
+                // eslint-disable-next-line no-continue
+                continue
+            }
+
+            currentTableLines = options.returnFromFunction ? [] : null
+
+            if (retTables.length > 0) {
+                // add a newline before the next header to pad the dumps
+                saveChunk('')
+            }
+
+            if(options.verbose) {
+                // write the table header to the file
+                const header = [
+                    '# ------------------------------------------------------------',
+                    `# DATA DUMP FOR TABLE: ${table.name}${options.lockTables ? ' (locked)' : ''}`,
```

Comment:
I added this `(locked)` indication here; not sure, how else I should verify that the flag took effect...
actually, this is still not completely true, since this not really actively checks that the lock was in effect, but at least the code coverage of 100% is giving the missing confidence...